### DISCUSSION TITLE
Wake on Lan under Debian Bullseye

### DISCUSSION THREAD

**Discussion by bwicksall** [ltsp/ltsp#666] [Q&A] [answered]
I have some Dell clients with Realtek network adapters.  The BIOS is correctly configured for WOL and they work fine as LTSP 5 clients under Debian Squeeze.  I'm now migrating to Bullseye with LTSP 21.  Under Bullseye the LAN interface is being turned off completely at shutdown so it will not receive the magic packet.

I can remote into a client and run the following to enable WOL:

ethtool -s enp3s0 wol g

Now if I shutdown the client it will respond to WOL packets.  Of course it only works once.  Is there a way I could automate this with LTSP so it works for all clients?  

Any possibility of adding parameters to /etc/network/interfaces for each client?  It looks like the following might work:

```
auto eth0
iface eth0 inet dhcp
        ethernet-wol g
```

**Comment by bwicksall** [comment]
Replying to myself.  I edited /usr/share/ltsp/client/init/45-networking.sh with the following:

```
--- 45-networking.sh.before     2022-02-08 21:17:27.190432443 -0500
+++ 45-networking.sh    2022-02-08 21:11:58.503779840 -0500
@@ -123,6 +123,7 @@
 
 auto %s
 iface %s inet manual
+        ethernet-wol g
 " "$DEVICE" "$DEVICE"> /etc/network/interfaces
 
     # Never ifdown anything. Safer!
```

Now my clients wake properly.

**Comment by alkisg** [comment, marked as answer]
Do you have epoptes-client installed?
In previous Debian versions, epoptes-client shipped /etc/default/epoptes-client, which contained the following:

```
# The server where epoptes-client will be connecting to.
# If unset, thin client user sessions running on the server will try to connect
# to "localhost", while thin client root sessions and fat or standalone clients
# will try to connect to "server".
# LTSP automatically puts "server" in /etc/hosts for thin and fat clients,
# but you'd need to put "server" in DNS manually for standalone clients.
#SERVER=server

# The port where the server will be listening on, and where the client will try
# to connect to. For security reasons it defaults to a system port, 789.
#PORT=789

# Set Wake On LAN for devices that support it. Comment it out to disable it.
WOL=g
```

In bullseye this was omitted by a mistake, so WOL=g wasn't activated.
In testing, that file is back in its place. You can just copy the contents from here and create it yourself and it should then work with no modifications in LTSP.

**Reply by bwicksall** [reply]
I can confirm that adding the epoptes-client file does indeed work.  As always Thanks!
